Plugin crashed

<img width="1424" height="723" alt="Image" src="https://github.com/user-attachments/assets/353cae1f-cc2d-4766-af73-01b5293b5fee" />

Stack trace from the terraform-provider-proxmox_v3.0.2-rc01.exe plugin:

panic: assignment to entry in nil map

goroutine 73 [running]:
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk.terraform_Disk_QemuScsiStorage(0xc000717ef0, 0xc00053eb5e, 0x0)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go:302 +0xed
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk.terraform_Disk_QemuScsiDisks(0xc00053ea98?, 0xc00053eb5e, 0xc00053ea98)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go:198 +0x15d
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk.terraform_Disk_QemuDisks({0x0?, 0x0?, 0xc001010300?, 0x0?}, 0xc00053eb5e, 0xc00053ea98)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go:28 +0x255
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk.Terraform_Unsafe(0xc0006db710, 0xc000986500, 0xc00053eb5e)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk/terraform.go:11 +0x233
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceVmQemuRead({0x17b3e18, 0xc000994690}, 0xc0006db710, {0x14a0100?, 0xc0006e6aa0})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:1079 +0x7aa
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000147700, {0x17b3d70, 0xc000888ed0}, 0xc0006db710, {0x14a0100, 0xc0006e6aa0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:866 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000147700, {0x17b3d70, 0xc000888ed0}, 0xc0009eed00, {0x14a0100, 0xc0006e6aa0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:1162 +0x52a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0002f2000, {0x17b3d70?, 0xc000888e10?}, 0xc0003f6a00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/grpc_provider.go:908 +0xb6d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0001f8140, {0x17b3d70?, 0xc000888300?}, 0xc000994070)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/tf5server/server.go:859 +0x2f3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x16288a0, 0xc0001f8140}, {0x17b3d70, 0xc000888300}, 0xc00055a400, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:651 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001dc600, {0x17b3d70, 0xc000888270}, 0xc00019e2a0, 0xc00052a7e0, 0x1e38938, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc0001dc600, {0x17b4730, 0xc000328000}, 0xc00019e2a0)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 27
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x11d

Error: The terraform-provider-proxmox_v3.0.2-rc01.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
@@ -100,7 +100,7 @@ func terraform_Disk_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool, 
 		schema[schemaReplicate] = config.Passthrough.Replicate
 		schema[schemaSerial] = string(config.Passthrough.Serial)
 		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
-		schema[schemaType] = schemaDisk
+		schema[schemaType] = schemaPassthrough
 		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {

--- a/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
@@ -100,7 +100,7 @@ func terraform_Disk_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool, 
 		schema[schemaReplicate] = config.Passthrough.Replicate
 		schema[schemaSerial] = string(config.Passthrough.Serial)
 		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
-		schema[schemaType] = schemaPassthrough
+		schema[schemaType] = schemaDisk // yes, passthrough is a disk type
 		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {
@@ -178,7 +178,7 @@ func terraform_Disk_QemuSataStorage(config *pveAPI.QemuSataStorage, ciDisk *bool
 		schema[schemaReplicate] = config.Passthrough.Replicate
 		schema[schemaSerial] = string(config.Passthrough.Serial)
 		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
-		schema[schemaType] = schemaPassthrough
+		schema[schemaType] = schemaDisk // yes, passthrough is a disk type
 		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {
@@ -335,7 +335,7 @@ func terraform_Disk_QemuScsiStorage(config *pveAPI.QemuScsiStorage, ciDisk *bool
 		schema[schemaReplicate] = config.Passthrough.Replicate
 		schema[schemaSerial] = string(config.Passthrough.Serial)
 		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
-		schema[schemaType] = schemaPassthrough
+		schema[schemaType] = schemaDisk // yes, passthrough is a disk type
 		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {
@@ -445,7 +445,7 @@ func terraform_Disk_QemuVirtIOStorage(config *pveAPI.QemuVirtIOStorage, schema m
 		schema[schemaReplicate] = config.Passthrough.Replicate
 		schema[schemaSerial] = string(config.Passthrough.Serial)
 		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
-		schema[schemaType] = schemaPassthrough
+		schema[schemaType] = schemaDisk // yes, passthrough is a disk type
 		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {

--- a/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
@@ -64,7 +64,10 @@ func terraform_Disk_QemuIdeDisks(config *pveAPI.QemuIdeDisks, ciDisk *bool, sche
 }
 
 func terraform_Disk_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool, schema map[string]any) map[string]any {
-	if schema != nil && schema[schemaType] == enumIgnore {
+	if schema == nil {
+		return nil
+	}
+	if schema[schemaType] == enumIgnore {
 		return schema
 	}
 	if config == nil {
@@ -142,7 +145,10 @@ func terraform_Disk_QemuSataDisks(config *pveAPI.QemuSataDisks, ciDisk *bool, sc
 }
 
 func terraform_Disk_QemuSataStorage(config *pveAPI.QemuSataStorage, ciDisk *bool, schema map[string]any) map[string]any {
-	if schema != nil && schema[schemaType] == enumIgnore {
+	if schema == nil {
+		return nil
+	}
+	if schema[schemaType] == enumIgnore {
 		return schema
 	}
 	if config == nil {
@@ -295,7 +301,10 @@ func terraform_Disk_QemuScsiDisks(config *pveAPI.QemuScsiDisks, ciDisk *bool, sc
 }
 
 func terraform_Disk_QemuScsiStorage(config *pveAPI.QemuScsiStorage, ciDisk *bool, schema map[string]any) map[string]any {
-	if schema != nil && schema[schemaType] == enumIgnore {
+	if schema == nil {
+		return nil
+	}
+	if schema[schemaType] == enumIgnore {
 		return schema
 	}
 	if config == nil {
@@ -407,7 +416,10 @@ func terraform_Disk_QemuVirtIODisks(config *pveAPI.QemuVirtIODisks, schema map[s
 }
 
 func terraform_Disk_QemuVirtIOStorage(config *pveAPI.QemuVirtIOStorage, schema map[string]any) map[string]any {
-	if schema != nil && schema[schemaType] == enumIgnore {
+	if schema == nil {
+		return nil
+	}
+	if schema[schemaType] == enumIgnore {
 		return schema
 	}
 	if config == nil {

--- a/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
@@ -89,8 +89,7 @@ func terraform_Disk_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool, 
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Disk.Bandwidth)
-	}
-	if config.Passthrough != nil {
+	} else if config.Passthrough != nil {
 		schema[schemaAsyncIO] = string(config.Disk.AsyncIO)
 		schema[schemaBackup] = config.Disk.Backup
 		schema[schemaCache] = string(config.Disk.Cache)
@@ -104,11 +103,9 @@ func terraform_Disk_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool, 
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
-	}
-	if config.CdRom != nil {
+	} else if config.CdRom != nil {
 		terraform_Disk_QemuCdRom_unsafe(config.CdRom, schema)
-	}
-	if config.CloudInit != nil {
+	} else if config.CloudInit != nil {
 		*ciDisk = true
 		terraform_Disk_QemuCloudInit_unsafe(config.CloudInit, schema)
 	}
@@ -170,8 +167,7 @@ func terraform_Disk_QemuSataStorage(config *pveAPI.QemuSataStorage, ciDisk *bool
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Disk.Bandwidth)
-	}
-	if config.Passthrough != nil {
+	} else if config.Passthrough != nil {
 		schema[schemaAsyncIO] = string(config.Disk.AsyncIO)
 		schema[schemaBackup] = config.Disk.Backup
 		schema[schemaCache] = string(config.Disk.Cache)
@@ -185,11 +181,9 @@ func terraform_Disk_QemuSataStorage(config *pveAPI.QemuSataStorage, ciDisk *bool
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
-	}
-	if config.CdRom != nil {
+	} else if config.CdRom != nil {
 		terraform_Disk_QemuCdRom_unsafe(config.CdRom, schema)
-	}
-	if config.CloudInit != nil {
+	} else if config.CloudInit != nil {
 		*ciDisk = true
 		terraform_Disk_QemuCloudInit_unsafe(config.CloudInit, schema)
 	}
@@ -328,8 +322,7 @@ func terraform_Disk_QemuScsiStorage(config *pveAPI.QemuScsiStorage, ciDisk *bool
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Disk.Bandwidth)
-	}
-	if config.Passthrough != nil {
+	} else if config.Passthrough != nil {
 		schema[schemaAsyncIO] = string(config.Disk.AsyncIO)
 		schema[schemaBackup] = config.Disk.Backup
 		schema[schemaCache] = string(config.Disk.Cache)
@@ -345,11 +338,9 @@ func terraform_Disk_QemuScsiStorage(config *pveAPI.QemuScsiStorage, ciDisk *bool
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
-	}
-	if config.CdRom != nil {
+	} else if config.CdRom != nil {
 		terraform_Disk_QemuCdRom_unsafe(config.CdRom, schema)
-	}
-	if config.CloudInit != nil {
+	} else if config.CloudInit != nil {
 		*ciDisk = true
 		terraform_Disk_QemuCloudInit_unsafe(config.CloudInit, schema)
 	}
@@ -442,8 +433,7 @@ func terraform_Disk_QemuVirtIOStorage(config *pveAPI.QemuVirtIOStorage, schema m
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Disk.Bandwidth)
-	}
-	if config.Passthrough != nil {
+	} else if config.Passthrough != nil {
 		schema[schemaAsyncIO] = string(config.Passthrough.AsyncIO)
 		schema[schemaBackup] = config.Passthrough.Backup
 		schema[schemaCache] = string(config.Passthrough.Cache)
@@ -458,8 +448,7 @@ func terraform_Disk_QemuVirtIOStorage(config *pveAPI.QemuVirtIOStorage, schema m
 		schema[schemaType] = schemaDisk
 		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
-	}
-	if config.CdRom != nil {
+	} else if config.CdRom != nil {
 		terraform_Disk_QemuCdRom_unsafe(config.CdRom, schema)
 	}
 	return schema

--- a/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/terraform_disk.go
@@ -90,18 +90,18 @@ func terraform_Disk_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool, 
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Disk.Bandwidth)
 	} else if config.Passthrough != nil {
-		schema[schemaAsyncIO] = string(config.Disk.AsyncIO)
-		schema[schemaBackup] = config.Disk.Backup
-		schema[schemaCache] = string(config.Disk.Cache)
-		schema[schemaDiscard] = config.Disk.Discard
-		schema[schemaEmulateSSD] = config.Disk.EmulateSSD
+		schema[schemaAsyncIO] = string(config.Passthrough.AsyncIO)
+		schema[schemaBackup] = config.Passthrough.Backup
+		schema[schemaCache] = string(config.Passthrough.Cache)
+		schema[schemaDiscard] = config.Passthrough.Discard
+		schema[schemaEmulateSSD] = config.Passthrough.EmulateSSD
 		schema[schemaFile] = config.Passthrough.File
 		schema[schemaPassthrough] = true
-		schema[schemaReplicate] = config.Disk.Replicate
-		schema[schemaSerial] = string(config.Disk.Serial)
-		schema[schemaSize] = size.String(int64(config.Disk.SizeInKibibytes))
+		schema[schemaReplicate] = config.Passthrough.Replicate
+		schema[schemaSerial] = string(config.Passthrough.Serial)
+		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
 		schema[schemaType] = schemaDisk
-		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
+		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {
 		terraform_Disk_QemuCdRom_unsafe(config.CdRom, schema)
@@ -168,18 +168,18 @@ func terraform_Disk_QemuSataStorage(config *pveAPI.QemuSataStorage, ciDisk *bool
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Disk.Bandwidth)
 	} else if config.Passthrough != nil {
-		schema[schemaAsyncIO] = string(config.Disk.AsyncIO)
-		schema[schemaBackup] = config.Disk.Backup
-		schema[schemaCache] = string(config.Disk.Cache)
-		schema[schemaDiscard] = config.Disk.Discard
-		schema[schemaEmulateSSD] = config.Disk.EmulateSSD
+		schema[schemaAsyncIO] = string(config.Passthrough.AsyncIO)
+		schema[schemaBackup] = config.Passthrough.Backup
+		schema[schemaCache] = string(config.Passthrough.Cache)
+		schema[schemaDiscard] = config.Passthrough.Discard
+		schema[schemaEmulateSSD] = config.Passthrough.EmulateSSD
 		schema[schemaFile] = config.Passthrough.File
 		schema[schemaPassthrough] = true
-		schema[schemaReplicate] = config.Disk.Replicate
-		schema[schemaSerial] = string(config.Disk.Serial)
-		schema[schemaSize] = size.String(int64(config.Disk.SizeInKibibytes))
-		schema[schemaType] = schemaDisk
-		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
+		schema[schemaReplicate] = config.Passthrough.Replicate
+		schema[schemaSerial] = string(config.Passthrough.Serial)
+		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
+		schema[schemaType] = schemaPassthrough
+		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {
 		terraform_Disk_QemuCdRom_unsafe(config.CdRom, schema)
@@ -323,20 +323,20 @@ func terraform_Disk_QemuScsiStorage(config *pveAPI.QemuScsiStorage, ciDisk *bool
 		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Disk.Bandwidth)
 	} else if config.Passthrough != nil {
-		schema[schemaAsyncIO] = string(config.Disk.AsyncIO)
-		schema[schemaBackup] = config.Disk.Backup
-		schema[schemaCache] = string(config.Disk.Cache)
-		schema[schemaDiscard] = config.Disk.Discard
-		schema[schemaEmulateSSD] = config.Disk.EmulateSSD
+		schema[schemaAsyncIO] = string(config.Passthrough.AsyncIO)
+		schema[schemaBackup] = config.Passthrough.Backup
+		schema[schemaCache] = string(config.Passthrough.Cache)
+		schema[schemaDiscard] = config.Passthrough.Discard
+		schema[schemaEmulateSSD] = config.Passthrough.EmulateSSD
 		schema[schemaFile] = config.Passthrough.File
-		schema[schemaIOthread] = config.Disk.IOThread
+		schema[schemaIOthread] = config.Passthrough.IOThread
 		schema[schemaPassthrough] = true
-		schema[schemaReadOnly] = config.Disk.ReadOnly
-		schema[schemaReplicate] = config.Disk.Replicate
-		schema[schemaSerial] = string(config.Disk.Serial)
-		schema[schemaSize] = size.String(int64(config.Disk.SizeInKibibytes))
-		schema[schemaType] = schemaDisk
-		schema[schemaWorldWideName] = string(config.Disk.WorldWideName)
+		schema[schemaReadOnly] = config.Passthrough.ReadOnly
+		schema[schemaReplicate] = config.Passthrough.Replicate
+		schema[schemaSerial] = string(config.Passthrough.Serial)
+		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
+		schema[schemaType] = schemaPassthrough
+		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {
 		terraform_Disk_QemuCdRom_unsafe(config.CdRom, schema)
@@ -445,7 +445,7 @@ func terraform_Disk_QemuVirtIOStorage(config *pveAPI.QemuVirtIOStorage, schema m
 		schema[schemaReplicate] = config.Passthrough.Replicate
 		schema[schemaSerial] = string(config.Passthrough.Serial)
 		schema[schemaSize] = size.String(int64(config.Passthrough.SizeInKibibytes))
-		schema[schemaType] = schemaDisk
+		schema[schemaType] = schemaPassthrough
 		schema[schemaWorldWideName] = string(config.Passthrough.WorldWideName)
 		terraformQemuDiskBandwidth(schema, config.Passthrough.Bandwidth)
 	} else if config.CdRom != nil {

